packages/block-editor/src/components/link-control/index.js

```diff
@@ -292,8 +292,6 @@ function LinkControl( {
 	const shownUnlinkControl =
 		onRemove && value && ! isEditingLink && ! isCreatingPage;
 
-	const showSettings = !! settings?.length;
-
```

Comment:
Question: why are we removing this? It's used as a check to ensure there are actually settings before rendering them. It is possible to pass `null` as settings or variations thereof. 

Note to self: add test coverage for this.

Reply:
Added in https://github.com/WordPress/gutenberg/pull/51712/commits/0408cbf170c322a38fce1e67acc08e30966894d7